.github/workflows/CI-python.yml

```diff
@@ -43,43 +39,57 @@ jobs:
         with:
           python-version: ${{ matrix.pythonVersion }}
 
-      - if: ${{ matrix.pythonVersion == '3.7' }}
-        name: Setup tools for python lt 3.8
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: ${{ matrix.pythonVersion }}
+
+      - if: ${{ matrix.operatingSystem != 'macos-latest' }}
+        name: Install pytorch on non-MacOS
+        shell: bash -l {0}
         run: |
-          python -m pip install --upgrade "pip<=23.1.2"
-          pip install --upgrade setuptools
-          pip install --upgrade "pip-tools<=6.13.0"
+          conda install --yes --quiet pytorch torchvision captum cpuonly -c pytorch
+
+      - if: ${{ matrix.operatingSystem == 'macos-latest' }}
+        name: Install Anaconda packages on MacOS, which should not include cpuonly according to official docs
+        shell: bash -l {0}
+        run: |
+          conda install --yes --quiet pytorch torchvision captum -c pytorch
 
-      - if: ${{ matrix.pythonVersion == '3.8' || matrix.pythonVersion == '3.9' || matrix.pythonVersion == '3.10' }}
-        name: Setup tools for python gte 3.8
+      - name: Setup tools
+        shell: bash -l {0}
         run: |
           python -m pip install --upgrade pip
           pip install --upgrade setuptools
           pip install --upgrade "pip-tools<=7.1.0"
 
-      - name: Pip compile
-        run: |
-          pip-compile requirements-dev.txt -o requirements-dev-comp.txt
-          cat requirements-dev-comp.txt
-        working-directory: ${{ matrix.packageDirectory }}
-
-      - name: Upload requirements
-        uses: actions/upload-artifact@v3
-        with:
-          name: requirements-dev-comp.txt
-          path: ${{ matrix.packageDirectory }}/requirements-dev-comp.txt
-
-      - name: Install dependencies
+      - name: Install package dependencies
+        shell: bash -l {0}
         run: |
-          pip-sync requirements-dev-comp.txt
+          pip install -r requirements-dev.txt
         working-directory: ${{ matrix.packageDirectory }}
 
       - name: Install package
+        shell: bash -l {0}
         run: |
           pip install -v -e .
         working-directory: ${{ matrix.packageDirectory }}
 
+      - name: Pip freeze
```

Comment:
Why is the raiwidgets logic added here??

Reply:
it's just to upload dependencies